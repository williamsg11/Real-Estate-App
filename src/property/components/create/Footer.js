import React, {PropTypes} from 'react';
import {StyleSheet, Text, TouchableHighlight} from 'react-native';
import colors from '../../../common/colors';

export class Footer extends React.Component {
  static propTypes = {
    title: PropTypes.string,
    updateListing: PropTypes.func.isRequired,
  };

  render() {
    const {title, updateListing, disabled = false} = this.props;
    return (
      <TouchableHighlight
        underlayColor="transparent"
        onPress={() => updateListing()}
        style={[styles.container, disabled && {opacity: 0.5}]}
        disabled={disabled}>
        <Text style={styles.text}>
          {title ? title : 'Next'}
        </Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.tomato,
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
    height: 56,
    alignItems: 'center',
    justifyContent: 'center',
    borderColor: 'transparent',
  },
  text: {
    color: 'white',
    fontWeight: '500',
    fontSize: 18,
  },
});

export default Footer;
