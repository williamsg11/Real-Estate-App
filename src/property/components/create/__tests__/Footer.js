import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './../Footer';

test('renders Footer', () => {
  const tree = renderer.create(<Footer updateListing={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
