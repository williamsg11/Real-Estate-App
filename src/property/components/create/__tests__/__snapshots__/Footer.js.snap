// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#d60c3f",
          "borderColor": "transparent",
          "bottom": 0,
          "height": 56,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        false,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "500",
      }
    }
  >
    Next
  </Text>
</View>
`;
